catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SetVariableTest.java

```diff
@@ -0,0 +1,170 @@
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.hamcrest.core.Is;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.longClick;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withChild;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withParent;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.hamcrest.Matchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class SetVariableTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	public void createProject() {
+		SetVariableBrick setVariableTestBrick1 = new SetVariableBrick(10);
+		Script script = BrickTestUtils.createProjectAndGetStartScript("setVariableTest");
+		script.addBrick(setVariableTestBrick1);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testCreateNewUserVariableAndDeletion() {
+		String userVariableName = "testVariable1";
+		String secondUserVariableName = "testVariable2";
+
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_set_variable);
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).performNewVariableInitial(userVariableName);
+		// manual perform new variable because of CAT-2359
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).perform(click());
+		onView(withText(R.string.brick_variable_spinner_create_new_variable))
+				.perform(click());
+		onView(withId(R.id.dialog_formula_editor_data_name_edit_text)).perform(typeText(secondUserVariableName));
+		onView(withId(android.R.id.button1)).perform(click());
+
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_set_variable_edit_text).perform(click());
```

Comment:
since something really similar is done also in testDeleteItemOfUserListBrickMultipleLists,
maybe put deleting an item of the data list in formula editor into a function.